src/main/java/org/spongepowered/api/entity/player/Player.java

```diff
@@ -94,6 +96,51 @@
     void kick(Text reason);
 
     /**
+     * Edits a sign on this {@link Player}'s client.
+     *
+     * <p>Any pre-existing lines on the {@link Sign} given will be filled in to the
+     * dialog, and the {@link Sign} will be edited with the contents of the dialog
+     * after the "Done" button is pressed by the player.</p>
+     *
+     * <p>This method does *not* block until the player has finished editing</p>
+     *
+     * @param sign The sign to edit
+     */
+    void editSign(SignData sign);
```

Comment:
`SignData` does not contain a link to a real `Sign` TileEntity so i guess this should be just a `Sign` or a `Location`.